lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -393,6 +394,60 @@ def set_information(self, trans, id, payload={}, **kwd):
         return {'message': 'User information has been saved.'}
 
+    def send_verification_email( self, trans, email, username ):
+        """
+        Send the verification email containing the activation link to the user's email.
+        """
+        if username is None:
+            username = trans.user.username
+        activation_link = self.prepare_activation_link( trans, escape( email ) )
+
+        host = trans.request.host.split( ':' )[ 0 ]
+        if host in [ 'localhost', '127.0.0.1', '0.0.0.0' ]:
+            host = socket.getfqdn()
+        body = ("Hello %s,\n\n"
+                "In order to complete the activation process for %s begun on %s at %s, please click on the following link to verify your account:\n\n"
+                "%s \n\n"
+                "By clicking on the above link and opening a Galaxy account you are also confirming that you have read and agreed to Galaxy's Terms and Conditions for use of this service (%s). This includes a quota limit of one account per user. Attempts to subvert this limit by creating multiple accounts or through any other method may result in termination of all associated accounts and data.\n\n"
+                "Please contact us if you need help with your account at: %s. You can also browse resources available at: %s. \n\n"
+                "More about the Galaxy Project can be found at galaxyproject.org\n\n"
+                "Your Galaxy Team" % (escape( username ), escape( email ),
+                                      datetime.utcnow().strftime( "%D"),
+                                      trans.request.host, activation_link,
+                                      trans.app.config.terms_url,
+                                      trans.app.config.error_email_to,
+                                      trans.app.config.instance_resource_url))
+        to = email
+        frm = trans.app.config.email_from or 'galaxy-no-reply@' + host
+        subject = 'Galaxy Account Activation'
+        try:
+            util.send_mail( frm, to, subject, body, trans.app.config )
+            return True
+        except Exception:
+            log.exception( 'Unable to send the activation email.' )
+            return False
+
+    def prepare_activation_link( self, trans, email ):
+        """
+        Prepare the account activation link for the user.
+        """
+        activation_token = self.get_activation_token( trans, email )
+        activation_link = url_for( controller='user', action='activate', activation_token=activation_token, email=email, qualified=True  )
+        return activation_link
+
+    def get_activation_token( self, trans, email ):
+        """
+        Check for the activation token. Create new activation token and store it in the database if no token found.
+        """
+        user = trans.sa_session.query( trans.app.model.User ).filter( trans.app.model.User.table.c.email == email ).first()
+        activation_token = user.activation_token
```

Comment:
Maybe we can reset the activation token to `None` after the activation has been completed.